File: minimum-changes-to-make-alternating-binary-string.cc
class Solution {
public:
  int minOperations(string s) {
    int N = s.size();

    int ans = N;
    for (int f : {0, 1}) {
      int cur = 0;
      for (char c : s) {
        int ci = c - '0';
        cur += (ci != f);
        f ^= 1;
      }
      ans = min(ans, cur);
    }
    return ans;
  }
};

File: m/minimum-changes-to-make-alternating-binary-string.erl
-spec min_operations(S :: unicode:unicode_binary()) -> integer().
min_operations(S) ->
  min(rec(S, 0), rec(S, 1)).

rec(<<>>, _T) ->
  0;
rec(<<Head, Rest/binary>>, T) ->
  case Head =:= T + $0 of
    true -> rec(Rest, 1 - T);
    false -> rec(Rest, 1 - T) + 1
  end.
